python/tskit/metadata.py

```diff
@@ -603,6 +672,93 @@ def decode(self, encoded: bytes) -> Any:
         # Set by __init__
         pass  # pragma: nocover
 
+    def numpy_dtype(self, schema):
+        # Mapping from struct format characters to NumPy dtype strings
+        # Note: All are little-endian as enforced by the struct codec
```

Comment:
So given that we're forcing little endian, these all have the "standard size". We should note that.

Reply:
Not sure I get what you mean here - they would be the same size if they were big-endian?

Reply:
Format characters (https://github.com/tskit-dev/tskit/pull/3091#discussion_r1976098928) by default are platform dependent, so ``i`` means a C int, which can be anything from 2 to 8 bytes depending on your platform. By explicitly specifying "<i" struct uses the "standard size" so it's guaranteed to be 4 bytes.